### PR TITLE
Conf dir

### PR DESCRIPTION
**Pull request by davidedmundson**
Load configuration information from config directories

Our configuration generally consists of user set config options and
distro defined options.

For example, a distro might want to specify a custom theme, but the
user's autostart name can't come from the distro.

Using the same config file leads to problems when upgrading.

This loads configuration from all files in a directory
/usr/lib/sddm/conf.d then /etc/sddm/conf.d as well as the main config file /etc/sddm.conf

### CODE REVIEW

**Comment by shadeslayer** [comment]
Bump?

File: CMakeLists.txt [hunk 1]
```diff
@@ -160,6 +160,8 @@ set(SESSION_COMMAND             "${DATA_INSTALL_DIR}/scripts/Xsession"
 set(WAYLAND_SESSION_COMMAND     "${DATA_INSTALL_DIR}/scripts/wayland-session"       CACHE PATH      "Script to execute when starting the Wayland desktop session")
 
 set(CONFIG_FILE                 "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf"        CACHE PATH      "Path of the sddm config file")
+set(CONFIG_DIR                  "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf.d"      CACHE PATH      "Path of the sddm config directory")
+set(SYSTEM_CONFIG_DIR           "${CMAKE_INSTALL_PREFIX}/lib/sddm/sddm.conf.d"     CACHE PATH      "Path of the system sddm config directory")
```
Comment (pkulev) [inline CMakeLists.txt]: one space before CACHE

File: src/common/Constants.h.in [hunk 2]
```diff
@@ -33,6 +33,9 @@
 #define WAYLAND_SESSION_COMMAND     "@WAYLAND_SESSION_COMMAND@"
 
 #define CONFIG_FILE                 "@CONFIG_FILE@"
+#define CONFIG_DIR                 "@CONFIG_DIR@"
```
Comment (pkulev) [inline src/common/Constants.h.in]: one extra space here and below

File: src/common/ConfigReader.cpp [hunk 3]
```diff
@@ -133,21 +133,57 @@ namespace SDDM {
...
     }
 
-    void ConfigBase::load() {
-        // first check if there's at least anything to read, otherwise stick to default values
-        if (!QFile::exists(m_path))
-            return;
+    void ConfigBase::load()
+    {
+        //order of priority from least influence to most influence, is
+        //m_sysConfigDir in alphabetical order then m_configDir then m_path
```
Comment (plfiorini) [inline src/common/ConfigReader.cpp]: expand on what m_sysConfigDir and m_configDir and m_path are so it's more clear

Comment (plfiorini) [inline src/common/ConfigReader.cpp]: Something like Order of priority from least influence to most influence is system config dir, user config dir, legacy file

**Pull request merged**